core/src/main/java/org/aerogear/mobile/core/http/OkHttpServiceModule.java

```diff
@@ -21,25 +16,39 @@
      * This is the default no argument constructor for all ServiceModules.
      */
     public OkHttpServiceModule() {
-        this.httpServiceConfiguration = new HttpServiceConfiguration();
-        OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
-        this.client = clientBuilder.build();
+        this(new OkHttpClient.Builder().build());
     }
 
     /**
      * This constructer uses a specific client for manual configurations and testing.
+     *
      * @param client a default OkHttpClient instance to use.
      */
     public OkHttpServiceModule(OkHttpClient client) {
+        this(client, new HttpServiceConfiguration());
+    }
+
+    public OkHttpServiceModule(HttpServiceConfiguration serviceConfiguration) {
+        this(new OkHttpClient.Builder().build(), serviceConfiguration);
+    }
+
+    public OkHttpServiceModule(OkHttpClient client, HttpServiceConfiguration serviceConfiguration) {
         this.client = client;
+        this.httpServiceConfiguration = serviceConfiguration;
     }
 
     @Override
-    public void bootstrap(MobileCore core, ServiceConfiguration configuration) {
-        this.httpServiceConfiguration = new HttpServiceConfiguration(configuration);
-        OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
+    public String type() {
+        return "http";
+    }
 
-        this.client = clientBuilder.build();
+    @Override
+    public void configure(ServiceConfiguration serviceConfiguration) {
+        this.httpServiceConfiguration = new HttpServiceConfiguration(serviceConfiguration);
```

Comment:
We should instanciate a client here as well.  There may be information on the ServiceConfiguration which is relevant to OKHTTP